sklearn/ensemble/_voting.py

```diff
@@ -348,16 +349,14 @@ def _collect_probas(self, X):
         """Collect results from clf.predict calls."""
         return np.asarray([clf.predict_proba(X) for clf in self.estimators_])
 
-    def _predict_proba(self, X):
-        """Predict class probabilities for X in 'soft' voting."""
-        check_is_fitted(self)
-        avg = np.average(
-            self._collect_probas(X), axis=0, weights=self._weights_not_none
-        )
-        return avg
+    def _check_voting(self):
+        if self.voting == "hard":
+            raise AttributeError(
+                "predict_proba is not available when voting=%r" % self.voting
+            )
```

Comment:
We should return `True` if when the check is fine.
```suggestion
            )
            return True
```

Reply:
Fixed!